### DISCUSSION TITLE
Multi-Schema / Subsystems

### DISCUSSION THREAD

**Discussion by afflom** [emporous/community#11] [Design]
UOR is a Conceptual Model of an information system. Within the UOR Concept  there are subsystems that manage the behavior and validation of UOR Collections. Four UOR subsystems have been identified and UOR subsystem types are extensible. The known subsystem types are:

- [Content](#content)
- [Trust](#trust)
- [Data Policy](#data-policy)
- [Network Policy](#network-policy) 

## Multi-Schema
Multi-Schema is the idea that a UOR Collection can have more than one schema. When a UOR Collection is called, each subsystem of  UOR is optionally invoked to interact with the called Collection.

## Content 
The content subsystem references the default schema of a collection. The Content schema pertains to the primary function of the Collection. When a user refers to a Collection, they would usually be intending to interact with the Collection's Content Schema. 

## Trust
The Trust subsystem pertains to the management of trust of Collections. This includes signature validation and attestation/provenance checking. 

## Data Policy
Policies can be set on content such as ABAC ACLs, content expiration, subscription enforcement.

## Network Policy
Policies can be set on content as it is sent "over the wire" such as QoS, network ACLs, and other routing behavior. 

## Identity / Interaction model 

When a caller calls a collection, the behavior of the target and the interaction between the caller and the target is determined by the user's identity.

**Comment by antweiss** [comment]
This is very concise!
Not sure if this is part of the Content subsystem but what about context/environment requirements? Things like resource requests, runtime type, platform architecture, etc?

**Reply by antweiss** [reply]
I'll do my best to go over the other discussions - I'm sure it will spill more light.
